Add Zrms column to Dorothea's output

gonzaponte opened the pull request:
This PR includes a new column in the output DST produced by `Dorothea`.  This column is the standard deviation of the peak in time, providing another useful variable for the analysis. It is correlated with the `S2width` column, but not the same. Change summary:

- The `kdst` produced by `Dorothea` now contains another column (`Zrms`).
- The `Peak` class now contains a new attribute (`rms`) and a new method (`rms_above_threshold`).
- Cosmetics.

gonzaponte committed:
Add Zrms to table description

gonzaponte committed:
Add Zrms to KrEvent

gonzaponte committed:
Add method to compute the peak's rms

gonzaponte committed:
Check rms value in tests

gonzaponte committed:
Cosmetics

gonzaponte committed:
PEP8-ify spacing

gonzaponte committed:
Use class methods to avoid code repetition

gonzaponte committed:
Add global rms attribute to Peak

gonzaponte committed:
Include Zrms in base city

gonzaponte committed:
Update test file and fixture with new column

gonzaponte committed:
Improve peak composite

So far, the time array could contain multiple
times the same value, which is non-physical.

abotas commented on invisible_cities/evm/pmaps_test.py:
```diff
@@ -43,6 +44,13 @@ def s2si_input(draw, min_size=1, max_size=10):
+    mean = np.average(x, weights=y)
+    var  = np.average((x-mean)**2, weights=y)
+    return var**0.5
+
```
Can we not use `np.var`? (also in `xy-algorithms`?)

Also, since we do this in a couple places, maybe we can put an rms function in `invisible_cities.core`

gonzaponte replied:
`np.var` does not accept weights, that is why we use this `np.average` formulation.

abotas commented on invisible_cities/evm/pmaps.pyx:
```diff
@@ -60,6 +61,17 @@ cdef class Peak:
+        mean = np.average( t         , weights=E)
+        var  = np.average((t-mean)**2, weights=E)
+        return var**0.5
+
```
It seems like we are repeating some code -- i want to reiterate that maybe we can put some mean+var function in core (if we can't just use np.var)

gonzaponte replied:
Sounds good.

gonzaponte committed:
Add functions to compute weighted mean, var and std

gonzaponte committed:
Add first test to `weighted_mean_and_var`

This test sets all weights to a random value, but common to all data
points. This must be equivalent to unweighted data.

gonzaponte committed:
Add second test to `weighted_mean_and_var`

This test ensures that the average and variance of samples from an
analytical funtion are close to those of the function.

gonzaponte committed:
Add third test to `weighted_mean_and_var`

This test ensured that the frequentist interpretation of the unbiased
estimator of the variance behaves correctly.

gonzaponte committed:
Add fourth test to `weighted_mean_and_var`

This test ensures the unbiased estimator of the variance when
interpreted as reliability weights behaves correctly.

gonzaponte committed:
Propagate use of new functions to the rest of the code
